llvm/lib/CodeGen/SelectionDAG/TargetLowering.cpp

```diff
@@ -10225,8 +10225,10 @@ SDValue TargetLowering::LowerToTLSEmulatedModel(const GlobalAddressSDNode *GA,
 
   ArgListTy Args;
   ArgListEntry Entry;
-  std::string NameString = ("__emutls_v." + GA->getGlobal()->getName()).str();
-  Module *VariableModule = const_cast<Module*>(GA->getGlobal()->getParent());
+  const GlobalValue *GV = GA->getGlobal();
+  GV = GV->getAliaseeObject() ? GV->getAliaseeObject() : GV;
+  std::string NameString = ("__emutls_v." + GV->getName()).str();
+  Module *VariableModule = const_cast<Module*>(GV->getParent());
```

Comment:
Shouldn't need const_cast

Reply:
If the 'const_cast' was removed, there will be an error caused:

Reply:
![image](https://github.com/user-attachments/assets/9c18740f-ee81-414a-bb05-5255081f5501)

Reply:
Yes, you should also make VariableModule const

Reply:
done